src/lib/gossip_net/gossip_net.ml

```diff
@@ -234,6 +225,18 @@ module Make (Message : Message_intf) : S with type msg := Message.msg = struct
     | Error _ ->
         failwith "Could not restart Kademlia"
 
+  let unmark_all_disconnected_peers t =
+    Logger.info t.logger ~module_:__MODULE__ ~location:__LOC__
+      !"Clearing disconnected peer set : %{sexp: Peer.t list}"
+      (Hash_set.to_list t.disconnected_peers) ;
+    let disconnected_peers =
+      List.map
+        (Hash_set.to_list t.disconnected_peers)
+        ~f:Peer.to_communications_host_and_port
+    in
+    Hash_set.clear t.disconnected_peers ;
+    restart_kademlia t disconnected_peers
```

Comment:
do you have to call `record_peer_events` again here, since the new membership has its own events pipe?